#      A F&B manager wants to determine whether there is any significant difference in the diameter of the cutlet between two units. A randomly selected sample of cutlets was collected from both units and measured? Analyze the data and draw inferences at 5% significance level. Please state the assumptions and tests that you carried out to check validity of the assumptions.


In [14]:
import pandas as pd
import scipy
from scipy import stats
cutlets=pd.read_csv(r"C:\Users\disha\Downloads\cutlets.csv")
cutlets



Unit A  Unit B
0   6.8090  6.7703
1   6.4376  7.5093
2   6.9157  6.7300
3   7.3012  6.7878
4   7.4488  7.1522
5   7.3871  6.8110
6   6.8755  7.2212
7   7.0621  6.6606
8   6.6840  7.2402
9   6.8236  7.0503
10  7.3930  6.8810
11  7.5169  7.4059
12  6.9246  6.7652
13  6.9256  6.0380
14  6.5797  7.1581
15  6.8394  7.0240
16  6.5970  6.6672
17  7.2705  7.4314
18  7.2828  7.3070
19  7.3495  6.7478
20  6.9438  6.8889
21  7.1560  7.4220
22  6.5341  6.5217
23  7.2854  7.1688
24  6.9952  6.7594
25  6.8568  6.9399
26  7.2163  7.0133
27  6.6801  6.9182
28  6.9431  6.3346
29  7.0852  7.5459
30  6.7794  7.0992
31  7.2783  7.1180
32  7.1561  6.6965
33  7.3943  6.5780
34  6.9405  7.3875

In [15]:
#changing column names
cutlets.columns="Unit_A","Unit_B"

                 

In [18]:
#printing p values
print("p value of cutlet at unit A",stats.shapiro(cutlets.Unit_A))
print("p value of cutlet at unit B",stats.shapiro(cutlets.Unit_B))

p value of cutlet at unit A ShapiroResult(statistic=0.9649458527565002, pvalue=0.3199819028377533)
p value of cutlet at unit B ShapiroResult(statistic=0.9727300405502319, pvalue=0.5224985480308533)


P-value ≤ alpha: The data do not follow a normal distribution (Reject H0)
P-value > alpha: Cannot conclude the data do not follow a normal distribution (Fail to reject H0)

i. The hypothesis of Unit A :
H0 = Data is normally distributed
Ha = Data is not normally distributed
P-Value is 0.32. (0.32 > 0.05)
So data is normally distributed

ii. The hypothesis of Unit B :
H0 = Data is normally distributed
Ha = Data is not normally distributed
P-Value is 0.5225. (0.5225 > 0.05)
So data is normally distributed

In [20]:
#variance test
scipy.stats.levene(cutlets.Unit_A, cutlets.Unit_B)
print("variance",scipy.stats.levene(cutlets.Unit_A, cutlets.Unit_B))

variance LeveneResult(statistic=0.6650897638632386, pvalue=0.4176162212502553)


Ho= Variance of diameters of Unit A is equal to the variance of diameters of Unit B
Ha= Variance of diameters of Unit A is not equal to the variance of diameters of Unit B
P-Value is 0.3136
We fail to reject the Null hypothesis. So we will accept it and hence Variances of A is equal to Variances of B

In [23]:
#sample test
scipy.stats.ttest_ind(cutlets.Unit_A, cutlets.Unit_B)
print("sample t test values",scipy.stats.ttest_ind(cutlets.Unit_A, cutlets.Unit_B))

sample t test values Ttest_indResult(statistic=0.7228688704678063, pvalue=0.4722394724599501)


Ho= Averages of diameters of Unit A is equal to Averages of diameters of unit B
Ha= Averages of diameters of Unit A is not equal to Averages of diameters of unit B
P-value(0.4722)>0.05
The inference is that there is no significant difference in the diameters of Unit A and Unit B.

# A hospital wants to determine whether there is any difference in the average Turn Around Time (TAT) of reports of the laboratories on their preferred list. They collected a random sample and recorded TAT for reports of 4 laboratories. TAT is defined as sample collected to report dispatch.
   
  Analyze the data and determine whether there is any difference in average TAT among the different laboratories at 5% 

In [8]:
import pandas as pd
import scipy
from scipy import stats
lab=pd.read_csv(r"C:\Users\disha\Downloads\labTAT.csv")
print(lab)

     Laboratory 1  Laboratory 2  Laboratory 3  Laboratory 4
0          185.35        165.53        176.70        166.13
1          170.49        185.91        198.45        160.79
2          192.77        194.92        201.23        185.18
3          177.33        183.00        199.61        176.42
4          193.41        169.57        204.63        152.60
..            ...           ...           ...           ...
115        178.49        170.66        193.80        172.68
116        176.08        183.98        215.25        177.64
117        202.48        174.54        203.99        170.27
118        182.40        197.18        194.52        150.87
119        182.09        215.17        221.49        162.21

[120 rows x 4 columns]


In [9]:
lab.columns="Laboratory_1","Laboratory_2","Laboratory_3","Laboratory_4"
print(lab.columns)


Index(['Laboratory_1', 'Laboratory_2', 'Laboratory_3', 'Laboratory_4'], dtype='object')


In [10]:
#normality test
print("P value of Lab_1 data:", stats.shapiro(lab.Laboratory_1))
print("P value of Lab_2 data:", stats.shapiro(lab.Laboratory_2))
print("P value of Lab_3 data:", stats.shapiro(lab.Laboratory_3))


P value of Lab_1 data: ShapiroResult(statistic=0.9901824593544006, pvalue=0.5506953597068787)
P value of Lab_2 data: ShapiroResult(statistic=0.9936322569847107, pvalue=0.8637524843215942)
P value of Lab_3 data: ShapiroResult(statistic=0.9886345267295837, pvalue=0.4205053448677063)


1.P Value for Lab-1 is 0.4232 (As P>0.05, H0 is Valid. Data is normally distributed)
2.P Value for Lab-2 is 0.8637 (As P>0.05, H0 is Valid. Data is normally distributed)
3.P Value for lab-3 is 0.0654(As P>0.05, H0 is Valid. Data is normally distributed)
4.P Value for lab-4 is 0.6619 (As P>0.05, H0 is Valid. Data is normally distributed)

In [11]:
#variance test
print("variance:",scipy.stats.levene(lab.Laboratory_1,lab.Laboratory_2,lab.Laboratory_3,lab.Laboratory_4))

variance: LeveneResult(statistic=2.599642500418024, pvalue=0.05161343808309816)


In [12]:
 #anova test
F,p=stats.f_oneway(lab.Laboratory_1, lab.Laboratory_2, lab.Laboratory_3,lab.Laboratory_4)
print("oneway alvona result:",F,p)

oneway alvona result: 118.70421654401437 2.1156708949992414e-57


Ho= Average TAT for all the samples is the same
Ha= Averages TAT for all the samples is not the same
Similarly, by doing for different lab combinations you can see that P-value is < 0.05
Hence there is a significant difference in the average TAT for all the labs

#       Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across region![image.png]

In [26]:
buyer=pd.read_csv(r"C:\Users\disha\Downloads\buyerratio.csv")
print(buyer)

  Observed Values  East  West  North  South
0           Males    50   142    131     70
1         Females   435  1523   1356    750


In [ ]:
table = [[50,142,131,70],
        [435,1523,1356,750]]

In [35]:
buyer_table=buyer.iloc[:,1:6]
buyer_table



East  West  North  South
0    50   142    131     70
1   435  1523   1356    750

In [36]:
buyer_table.values



array([[  50,  142,  131,   70],
       [ 435, 1523, 1356,  750]], dtype=int64)

In [38]:
val=stats.chi2_contingency(buyer_table)
val

(1.595945538661058,
 0.6603094907091882,
 3,
 array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
        [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]]))

In [39]:
type(val)

tuple

In [40]:
no_of_rows=len(buyer_table.iloc[0:2,0])
no_of_columns=len(buyer_table.iloc[0,0:4])
degree_of_f=(no_of_rows-1)*(no_of_columns-1)
print('Degree of freedom=',degree_of_f)

Degree of freedom= 3


In [41]:
expected_value=val[3]
expected_value

array([[  42.76531299,  146.81287862,  131.11756787,   72.30424052],
       [ 442.23468701, 1518.18712138, 1355.88243213,  747.69575948]])

In [42]:
from scipy.stats import chi2
chi_square=sum([(o-e)**2/e for o,e in zip(buyer_table.values,expected_value)])
chi_square_statestic=chi_square[0]+chi_square[1]
chi_square_statestic

1.5152956451130446

In [43]:
critical_value=chi2.ppf(0.95,3)
critical_value

7.814727903251179

In [45]:
if chi_square_statestic>=critical_value:
    print('dependent(reject H0)')
else:
    print('independent (fail to reject H0)')

independent (fail to reject H0)


In [46]:
p_value=1-chi2.cdf(chi_square_statestic,3)
p_value

0.6787446296467897

In [48]:
if p_value<=0.05:
    print('dependent (reject H0)')
else:
    print('independent(fail to reject H0)')

independent(fail to reject H0)


#     TeleCall uses 4 centers around the globe to process customer order forms. They audit a certain %  of the customer order forms. Any error in order form renders it defective and has to be reworked before processing.  The manager wants to check whether the defective %  varies by centre. Please analyze the data at 5% significance level and help the manager draw appropriate inferences


In [ ]:
import pandas as pd
from scipy import stats as stats
from scipy.stats import chi2_contingency
from scipy.stats import chi2

In [52]:
custom=pd.read_csv(r'C:\Users\disha\Downloads\Costomer+OrderForm.csv')
custom.head()

Phillippines   Indonesia       Malta       India
0   Error Free  Error Free   Defective  Error Free
1   Error Free  Error Free  Error Free   Defective
2   Error Free   Defective   Defective  Error Free
3   Error Free  Error Free  Error Free  Error Free
4   Error Free  Error Free   Defective  Error Free

In [55]:
print(custom['Phillippines'].value_counts(),custom['Indonesia'].value_counts(),custom['Malta'].value_counts(),custom['India'].value_counts())

Error Free    271
Defective      29
Name: Phillippines, dtype: int64 Error Free    267
Defective      33
Name: Indonesia, dtype: int64 Error Free    269
Defective      31
Name: Malta, dtype: int64 Error Free    280
Defective      20
Name: India, dtype: int64


In [60]:
obs=np.array([[271,267,269,280],[29,33,31,20]])
obs

array([[271, 267, 269, 280],
       [ 29,  33,  31,  20]])

In [69]:
import pandas as pd
import numpy as np
from scipy import stats
from scipy.stats import norm
from scipy.stats import chi2_contingency
chi2_contingency(obs) 
stat, p, dof, expected = chi2_contingency([[271,267,269,280],[29,33,31,20]])
print(stat)
print(p)
print(dof)
print(exp)

3.858960685820355
0.2771020991233135
3
[[8.86373823e-02 4.30777678e+01 1.47669879e+02 1.31892425e+02
  7.27712909e+01]
 [9.11362618e-01 4.42922232e+02 1.51833012e+03 1.35610758e+03
  7.48228709e+02]]


 As (p_value = 0.2771) > (α = 0.05); Accept Null Hypthesis i.e. Independence of categorical variables Thus, customer order forms defective % does not varies by centre.